app/models/course/announcement.rb

```diff
@@ -18,4 +19,19 @@ def set_defaults
   def valid_in_future?
     valid_from > DateTime.now
   end
+
+  # Get the number of unread announcements
+  def self.get_unread_number(current_user, course)
+    count = -1
+    if(!current_user.nil? && !course.nil?)
+      count = 0
+      unread_announcements = Course::Announcement.unread_by(current_user)
+      unread_announcements.each do |ann|
```

Comment:
Can we use something like <code>unread_announcements.where(course: @course).count </code> to achieve this ? ( Don't declare a count and manually increase it, it's in-efficient) 

Check here for more functions: http://guides.rubyonrails.org/active_record_querying.html